src/main/haskell/matching-logic/test/Kore/MatchingLogic/ProofSystem/OnePlusOne/ProofSystem.hs

```diff
@@ -0,0 +1,283 @@
+    VariableSubstitution (SubstitutedVariable v1) t (SubstitutingVariable v2) ->
+      VariableSubstitution <$> (SubstitutedVariable <$> var v1)
+                           <*> term t
+                           <*> (SubstitutingVariable <$> var v2)
+    ForallRule v t1 t2 -> ForallRule <$> var v <*> term t1 <*> term t2
+    Framing l pos h -> (flip Framing pos) <$> label l <*> hypothesis h
+    PropagateOr l pos t1 t2
+      -> (flip PropagateOr pos) <$> label l <*> term t1 <*> term t2
+    PropagateExists l pos v t
+      -> (flip PropagateExists pos) <$> label l <*> var v <*> term t
+    Existence v -> Existence <$> var v
+    Singvar v t path1 path2
+      -> (\v' t' -> Singvar v' t' path1 path2) <$> var v <*> term t
+    Proj1 t1 t2 -> Proj1 <$> term t1 <*> term t2
+    Proj2 t1 t2 -> Proj2 <$> term t1 <*> term t2
+    AndIntro t1 t2 -> AndIntro <$> term t1 <*> term t2
+    OrElim t1 t2 t3 -> OrElim <$> term t1 <*> term t2 <*> term t3
+    OrIntroL t1 t2 -> OrIntroL <$> term t1 <*> term t2
+    OrIntroR t1 t2 -> OrIntroR <$> term t1 <*> term t2
+
+-- | Lens focusing on the terms within a Rule.
+ruleTerms :: (Applicative f)
+          => (termA -> f termB)
+          -> (MLRule sort label var termA hyp
+              -> f (MLRule sort label var termB hyp))
+ruleTerms f = transformRule pure pure pure f pure
+
+-- | The 'MLRuleSig' synonym instantiates 'MLRule' to use
+-- the sorts, labels, and patterns from the 'IsSignature' instance 'sig'
+type MLRuleSig sig var {- hyp -}
+    = MLRule (Sort sig) (Label sig) var (WFPattern sig var) {- hyp -}
+
+dummyFormulaVerifier :: formula -> Either (Error MLError) ()
+dummyFormulaVerifier _ = return ()
+
+-- AST.fromWFPattern
+substVar :: (Eq sort, Eq var)
+         => sort
+         -> var
+         -> var
+         -> Pattern sort label var
+         -> Maybe (Pattern sort label var)
+substVar sort varFrom varTo pat = go pat
+  where
+    go pat@(Fix (AST.Forall _ sortBound varBound _))
+      | (sortBound,varBound) == (sort,varFrom) = Just pat
+      | (sortBound,varBound) == (sort,varTo) = Nothing
+    go pat@(Fix (AST.Variable sortVar varVar))
+      | (sortVar,varVar) == (sort,varFrom) = Just (Fix (AST.Variable sort varTo))
+    go (Fix pat) = Fix <$> traverse go pat
+
+-- | This checks the minimal proof system over Kore.AST patterns.
+instance (IsSignature sig, Eq (Sort sig), Eq (Label sig), Eq var) =>
+         ProofSystem MLError (MLRuleSig sig var) (WFPattern sig var) where
+  checkDerivation conclusion rule = case rule & ruleTerms %~ Just of
+      Propositional1 a b ->
+          expect $ a --> b --> a
+      Propositional2 a b c ->
+          expect $ (a --> b --> c) --> (a --> b) --> (a --> c)
+      Propositional3 a b ->
+          expect $ (notP' a --> notP' b) --> (b --> a)
+      ModusPonens a (ImpliesP s a' b) | a == a' ->
+          expect $ Just b
+      ModusPonens _ _ -> Left (Error [] "hypotheses have wrong form")
```

Comment:
Could you explain why this match produces an error?

Reply:
This line is when the hypotheses don't have the form `A->B` and `A`.